exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/RefreshMetadataHandler.java

```diff
@@ -64,6 +65,11 @@ public PhysicalPlan getPlan(SqlNode sqlNode) throws ValidationException, RelConv
       final SchemaPlus schema = findSchema(config.getConverter().getDefaultSchema(),
           refreshTable.getSchemaPath());
 
+      if (schema == null) {
+        return direct(false, "Storage plugin or workspace does not exist [%s]",
```

Comment:
Why not just throw a UserException? That way SqlLine or scripts know an error has occurred.